dom/src/MainDOMSource.ts

```diff
@@ -111,28 +116,36 @@ export class MainDOMSource implements DOMSource {
     };
   }
 
-  public elements(): MemoryStream<Element> {
-    let output$: Stream<Element | Array<Element>>;
+  public elements(): MemoryStream<Array<Element>> {
+    let output$: Stream<Array<Element>>;
     if (this._namespace.length === 0) {
-      output$ = this._rootElement$;
+      output$ = this._rootElement$.map(x => [x]);
     } else {
       const elementFinder = new ElementFinder(
         this._namespace,
         this._isolateModule,
       );
       output$ = this._rootElement$.map(el => elementFinder.call(el));
     }
-    const out: DevToolEnabledSource & MemoryStream<Element> = adapt(
+    const out: DevToolEnabledSource & MemoryStream<Array<Element>> = adapt(
       output$.remember(),
     );
     out._isCycleSource = this._name;
     return out;
   }
 
+  public element(): MemoryStream<Element> {
+    return this.elements().map(arr => arr[0]).filter(Boolean).remember();
```

Comment:
It's correct to reuse `elements()` here because it has the ElementFinder which we want to reuse, but before returning this remembered stream we need to set `_isCycleSource` as well as adapt() the output.